src/base/digest32.h

```diff
@@ -41,7 +41,6 @@ class Digest32
     using UnderlyingType = lt::digest32<N>;
 
     Digest32() = default;
-    Digest32(const Digest32 &other) = default;
```

Comment:
From Qt documentation:
>QHash's key and value data types must be **_assignable data types_**.

This type is used as key so I would add copy-assignment operator instead of removing copy-constructor.

Reply:
Actually this is just removing the declaration. Compiler will still generate the constructors by default.
See:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c20-if-you-can-avoid-defining-default-operations-do
https://en.cppreference.com/w/cpp/language/rule_of_three (Rule of zero section)

Reply:
BTW, if following the "Rule of three", then this class would need to add `Digest32::operator=(const UnderlyingType &nativeDigest)`. What do you think?

Reply:
>Actually this is just removing the (redundant) declaration. Compiler will still generate the constructors by default

AFAIK, compiler doesn't do this if there is at least one user-defined constructor. Isn't that already true?

Reply:
It doesn't fail to compile probably since it uses "emulation" of copy constructor via call to conversion constructor and conversion operator (i.e. when copying of `other` is needed it calls `Digest32(other.operator UnderlyingType())`).

Reply:
>if following the "Rule of three", then this class would need to add `Digest32::operator=(const UnderlyingType &nativeDigest)`. What do you think?

"Rule of three" is about **copy** constructor/assignment operator, isn't it?

Reply:
>AFAIK, compiler doesn't do this if there is at least one user-defined constructor. Isn't that already true?

I don't think so. With changes from this PR, the compiler still accept the following code (tested):
```c++
Digest32<160> a;
Digest32<160> b = a;
```

Reply:
>>if following the "Rule of three", then this class would need to add `Digest32::operator=(const UnderlyingType &nativeDigest)`. What do you think?
>
>"Rule of three" is about **copy** constructor/assignment operator, isn't it?

Ah right, nevermind then.

Reply:
Ok. Refreshed my knowledge. It was about implicit default constructor.